code/game/objects/effects/effect_system/effects_chem_smoke.dm

```diff
@@ -94,22 +98,21 @@
 
 
 /datum/effect_system/smoke_spread/chem/start(effect_range = 2)
-	set waitfor = FALSE
-
-	var/color = mix_color_from_reagents(chemholder.reagents.reagent_list)
-
-	for(var/x in 0 to 99)
-		for(var/i = 0, i < rand(2, 6), i++)
-			if(effect_range < 3)
-				new /obj/effect/particle_effect/chem_smoke/small(location, color)
-			else
-				new /obj/effect/particle_effect/chem_smoke(location, color)
-
-		if(x % 10 == 0) //Once every 10 ticks.
-			INVOKE_ASYNC(src, PROC_REF(SmokeEm), effect_range)
+	eff_colour = mix_color_from_reagents(chemholder.reagents.reagent_list)
+	eff_range = effect_range
+	START_PROCESSING(SSprocessing, src)
+
+/datum/effect_system/smoke_spread/chem/process()
+	process_count++
+	if(eff_range < 3)
+		new /obj/effect/particle_effect/chem_smoke/small(location, eff_colour)
+	else
+		new /obj/effect/particle_effect/chem_smoke(location, eff_colour)
 
-		sleep(1)
-	qdel(src)
+	INVOKE_ASYNC(src, PROC_REF(SmokeEm), eff_range)
+	if(process_count > max_process_count)
+		STOP_PROCESSING(SSprocessing, src)
+		qdel(src)
```

Comment:
does having the rate that this process's have any in game effect?

Reply:
It updates the visual a little slower but still does the same amount of injecting - I think